src/main/java/com/suse/salt/netapi/calls/LocalCall.java

```diff
@@ -101,6 +145,76 @@ public LocalCall(String functionName, Optional<List<?>> arg,
     }
 
     /**
+     * Calls this salt call via the async client and returns the results
+     * as they come in via the event stream.
+     *
+     * @param client SaltClient instance
+     * @param target the target for the function
+     * @param events the event stream to use
+     * @param cancel future to cancel the action
+     * @return a map from minion id to future of the result.
+     * @throws SaltException if anything goes wrong
+     */
+    public Map<String, CompletionStage<Result<R>>> callAsync(
```

Comment:
yes it would mean that. I think we will just provide several overloads of this method to cover different use cases.